### PR TITLE
httpd: modernize

### PR DESCRIPTION
**Pull request by avikivity**
Stop using at_exit() and app_template::run_deprecated(), both of which should be removed.

### CODE REVIEW

**Comment by tchaikov** [comment]
known issue: #1645

File: apps/httpd/main.cc [hunk 1]
```diff
@@ -80,11 +81,19 @@ int main(int ac, char** av) {
...
-    return app.run_deprecated(ac, av, [&] {
+    return app.run(ac, av, [&] {
         return seastar::async([&] {
             seastar_apps_lib::stop_signal stop_signal;
             auto&& config = app.configuration();
             httpd::http_server_control prometheus_server;
+            bool prometheus_started = false;
+
+            auto stop_prometheus = defer([&] () noexcept {
+                if (prometheus_started) {
```
Comment (tchaikov) [inline apps/httpd/main.cc:92]: `sharded<Service>::stop()` is a no-op. so probably we can dispense with this `prometheus_started` flag and use `deferred_stop` instead?

Comment (avikivity) [reply apps/httpd/main.cc:92]: It's not clean to stop() something that isn't started and rely on implementation details.

Comment (tchaikov) [reply apps/httpd/main.cc:92]: can we document this behavior in `shared<>` that `sharded<Service>::stop()` is a no-op if the sharded service is not started at all? so it won't be implementation details anymore, but a feature. just like `delete null` is a no-op.

Comment (avikivity) [reply apps/httpd/main.cc:92]: IMO it encourages bad coding practices.

**Review by tchaikov — APPROVED** [review]

**Comment by avikivity** [comment]
@nyh or @xemul please review

**Review by nyh — APPROVED** [review]

**Pull request closed without merging**